tls/s2n_config.c

```diff
@@ -235,6 +243,32 @@ struct s2n_config *s2n_config_new(void)
     return new_config;
 }
 
+int s2n_config_init_session_ticket_keys(struct s2n_config *config)
+{
+    if (config->ticket_keys == NULL) {
+        notnull_check(config->ticket_keys = s2n_array_new(sizeof(struct s2n_ticket_key)));
+    }
+
+    if (config->ticket_key_hashes == NULL) {
+        notnull_check(config->ticket_key_hashes = s2n_array_new(SHA_DIGEST_LENGTH));
+    }
+
+    return 0;
+}
+
+int s2n_config_free_session_ticket_keys(struct s2n_config *config)
+{
+    if (config->ticket_keys != NULL) {
+        GUARD(s2n_array_free(config->ticket_keys));
```

Comment:
config->ticket_keys = NULL after that, otherwise s2n might crash on calling s2n_config_free_session_ticket_keys and then s2n_config_set_session_tickets_onoff(.., 0).

Reply:
Oh yeah.